### PR TITLE
feat(ngcc): expose `--create-ivy-entry-points` option on ivy-ngcc

### PR DESCRIPTION
**Pull request by petebacondarwin**
This allows a postinstall hook to generate the same
output as the CLI integration does.

See https://github.com/angular/angular/pull/32999#issuecomment-539937368

### CODE REVIEW

**Review by filipesilva — APPROVED** [review]

**Review by alan-agius4 — APPROVED** [review]

File: packages/compiler-cli/ngcc/main-ngcc.ts [hunk 1]
```diff
@@ -45,6 +45,13 @@ if (require.main === module) {
                 'If specified then only the first matching package.json property will be compiled.',
             type: 'boolean'
           })
+          .option('create-new-entry-points', {
+            describe:
+                'If specified then new `ivy__` entry-points will be added to package.json rather than modifying the ones in-place.\n' +
```
Comment (gkalpak) [inline packages/compiler-cli/ngcc/main-ngcc.ts]: They are `*_ivy_ngcc` entry-points actually :grin:

File: packages/compiler-cli/ngcc/main-ngcc.ts [hunk 2]
```diff
@@ -45,6 +45,13 @@ if (require.main === module) {
...
+                'For this to work you need to have custom resolution set up (e.g. in webpack) to look for these new entry-points.\n' +
+                'The Angular CLI does this already, so it safe to use this option if the project is being built via the CLI.',
```
Comment (gkalpak) [inline packages/compiler-cli/ngcc/main-ngcc.ts]: Typo: `so it safe` --> `so it is safe`

**Review by gkalpak — APPROVED** [review]
👍

File: packages/compiler-cli/ngcc/main-ngcc.ts [hunk 3]
```diff
@@ -45,6 +45,13 @@ if (require.main === module) {
                 'If specified then only the first matching package.json property will be compiled.',
             type: 'boolean'
           })
+          .option('create-new-entry-points', {
```
Comment (IgorMinar) [inline packages/compiler-cli/ngcc/main-ngcc.ts]: "new" sounds like we are just going to make up stuff, have you considered "create-alternative-entry-points" or something similar that carries the meaning of duplicate/alternative/ivy-specific entry-points will be created due to this flag?

Comment (IgorMinar) [reply packages/compiler-cli/ngcc/main-ngcc.ts]: or also "preserve-original-entry-points" could be a good one.

**Review by IgorMinar — APPROVED** [review]
I'm suggesting a different flag name, but otherwise lgtm

File: packages/compiler-cli/ngcc/main-ngcc.ts (hunk 3 above)
Comment (petebacondarwin) [reply packages/compiler-cli/ngcc/main-ngcc.ts]: `create-ivy-entry-points`?

Comment (alxhub) [reply packages/compiler-cli/ngcc/main-ngcc.ts]: @petebacondarwin +1 on the name.

**Comment by IgorMinar** [comment]
lgtm

**Pull request closed without merging**